pay-api/src/pay_api/models/invoice.py

```diff
@@ -76,6 +76,20 @@ def find_by_id_and_payment_id(cls, identifier: int, pay_id: int):
         """Return a Invoice by id."""
         return cls.query.filter_by(payment_id=pay_id).filter_by(id=identifier).one_or_none()
 
+    @classmethod
+    def update_invoices_for_revenue_updates(cls, fee_distribution_id: int):
+        """Find and update all invoices using the distribution id."""
+        query = db.session.query(Invoice) \
+            .join(PaymentLineItem, PaymentLineItem.invoice_id == Invoice.id) \
+            .filter(PaymentLineItem.fee_distribution_id == fee_distribution_id)
+
+        invoices: [Invoice] = query.all()
+        for invoice in invoices:
+            if invoice.invoice_status_code == InvoiceStatus.PAID.value:
```

Comment:
shouldn't we do for Unpaid invoices as well? not sure if thats gonna happen for direct pay..but was thinking of user going to bambora screen and closing screen kind of edge cases... 
also should we have a date check as well? like may be 1 week or 1 month time ; will records before that have to be rectified in paybc?

Reply:
Unpaid invoices we can think about it. Didn't want to handle bunch of 404 from PayBC. The job to update stale payment will be 30 minutes, so we shouldn't see lot of those issues. 
 
The job will check the GL status before updating the revenue account. This feature, we are considering only for any error in GL code. If the use case is for new GL code, we should be creating new GL code and map the filing types to GL code.

Reply:
yup.. thats good ..